--- compiler/src/codegen/compiler.rs
+++ compiler/src/codegen/compiler.rs
@@ -1621,6 +1621,7 @@
             } else {
                 return Err(self.compile_error("Expect pattern."));
             }
+            self.advance()?; // Consumes the parsed literal token because literal() and number() and string() don't.

             self.emit_byte(Op::Eq as u8);
             let fail = self.emit_jump(Op::JumpIfFalse as u8);
