--- compiler/src/codegen/compiler.rs
+++ compiler/src/codegen/compiler.rs
@@ -1288,7 +1288,6 @@
             self.consume(Token::FatArrow, "Expect '=>' after pattern.")?;
             // Body
             if self.check(Token::LeftBrace) {
-                self.consume(Token::LeftBrace, "Expect '{' start of arm body.")?;
                 self.block()?;
                 self.emit_byte(Op::Unit as u8); // block doesn't leave a value, we push Unit as expression result
             } else if self.check(Token::Print) {
